plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1477,29 +1477,44 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding not null columns");
         }
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
-        icebergTable.updateSchema()
-                .addColumn(column.getName(), toIcebergType(column.getType()), column.getComment())
-                .commit();
+        try {
+            icebergTable.updateSchema()
+                    .addColumn(column.getName(), toIcebergType(column.getType()), column.getComment())
+                    .commit();
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, format("Failed to add column %s", column.getName()), e);
```

Comment:
Column name probably isn't needed in the exc msg, since it's obvious from the ALTER TABLE ADD COLUMN statement 
(otherwise we'd want to include table name as well)

OTOH it would be good to include `e`'s message if any, since it may contain important information.


```
throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to add column: " + firstNonNull(e.getMessage(), e), e);
```

Reply:
this will result in an error message like this: `Failed to add column: Failed to replace table due to concurrent updates: tpch.test_add_columnfyzkr8hk0k` 
if that's what we want then let me know and I'll make the adjustment

Reply:
I like that because it hints at the problem ("concurrent updates")

Not that top-level exception message serves two goals

- needs to be understood by a user
- if forwarded to a support engineer, needs to be informative and useful
  - in many BI tools, getting the full stack is hard 

so yes, please do

Reply:
done